### PR TITLE
Enhancement/import survey users

### PR DESCRIPTION
**Pull request by ramirobg94**
- Users
- Taxonomies
- Categories

### CODE REVIEW

File: test/e2e/tests/018-surveyImport.js [hunk 1]
```diff
@@ -1,11 +1,94 @@
-import { expectExists, fileSelect, getElement, reload, waitFor } from '../utils/api'
+import fs from 'fs'
```
Comment (minotogna) [inline test/e2e/tests/018-surveyImport.js:1]: please, split this into sub files

**Review by minotogna — CHANGES_REQUESTED** [review]

File: test/e2e/tests/018-surveyImport.js [hunk 2]
```diff
@@ -43,6 +126,193 @@ describe('Survey import', () => {
...
+
+    const taxonomyMockDataParsedByCode = taxonomyMockDataParsed.reduce(
+      (acc, taxon) => ({ ...acc, [taxon.code]: { ...taxon } }),
+      {}
+    )
+    const taxonomyTaxaByCode = taxonomy.reduce((acc, taxon) => ({ ...acc, [Taxon.getCode(taxon)]: { ...taxon } }), {})
+
+    await expect(Object.keys(taxonomyMockDataParsedByCode).sort()).toEqual(Object.keys(taxonomyTaxaByCode).sort())
+
+    await taxonomy.reduce(async (promise, taxon) => {
```
Comment (SteRiccio) [inline test/e2e/tests/018-surveyImport.js:208]: would `await PromiseUtils.each(taxonomy, async taxon => {...})` be more readable?

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: test/e2e/utils/surveyExport/categories.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  await expect(itemsRegionLevel.length).toBe(5)
+
+  const parent = itemsCountryLevel[0]
+  await PromiseUtils.each(itemsRegionLevel, async (item, index) =>
+    checkItemRegionLevel({ item, index, regionLevel, parentUuid: CategoryItem.getUuid(parent) })
+  )
+}
+
+// District level
+const checkItemDiscrictLevel = async ({ item, index, itemsRegionLevel, districtLevel }) => {
```
Comment (SteRiccio) [inline test/e2e/utils/surveyExport/categories.js:54]: just a typo... discrict...

File: test/e2e/utils/surveyExport/categories.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import * as CategoryLevel from '@core/survey/categoryLevel'
+import fs from 'fs'
+import path from 'path'
+import * as Category from '@core/survey/category'
+import * as CategoryItem from '@core/survey/categoryItem'
+import * as PromiseUtils from '@core/promiseUtils'
+
+import { checkFileAndGetContent } from './utils'
+
+const checkLevelAndReturnLevel = async ({ levels, categoryName, index }) => {
```
Comment (SteRiccio) [inline test/e2e/utils/surveyExport/categories.js]: categoryName must be renamed into levelName

**Review by minotogna — APPROVED** [review]

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**